### DISCUSSION TITLE
Time missmatch in tokens!!!!!????

### DISCUSSION THREAD

**Discussion by tamis-laan** [keycloak/keycloak#32271] [Q&A]
Hey,
I just refreshed my tokens (less then a second ago) and these are the values:

```
current time 1724154806

access token exp': 1724155097
access token iat': 1724154797

id token exp': 1724154300
id token iat': 1724154000
```

The access token and id token both have a different issued at (iat) value which I would expect to be the same as there issued at the same time???
And the id token has already expired as I receive it!!???
How can this be? Where does keycloak get it's time from? Both keycloak and my app are running in docker compose and both should receive my computers time. 

Could this be a bug?

**Comment by msanket79** [comment]
i am also facing a problem like this.
sometimes i am getting token issued into the past ,already expired tokens
what could be the reason?
Did you get it why it happening?
